### PR TITLE
add stat support

### PR DESCRIPTION
**Pull request by zhaoya881010**
add stat interface.

### CODE REVIEW

**Comment by corny** [comment]
We still need a test for this feature.

**Comment by zhaoya881010** [comment]
@corny It works well on FileZilla Server 0.9.60.

File: ftp.go [hunk 1]
```diff
@@ -535,6 +535,28 @@ func (c *ServerConn) List(path string) (entries []*Entry, err error) {
...
+func (c *ServerConn) Stat(path string)(entry *Entry, err error){
+	var cmd string
+	var parser parseFunc
+
+	if c.mlstSupported {
+		cmd = "MLST"
+		parser = parseRFC3659ListLine
+	} else {
+		return nil, errors.New("unexpected MLST response")
+	}
```
Comment (corny) [inline ftp.go]: Return return directly a `MLST unsupported` error message.

File: ftp.go [hunk 2]
```diff
@@ -535,6 +535,28 @@ func (c *ServerConn) List(path string) (entries []*Entry, err error) {
...
+	_, msg, err := c.cmd(StatusRequestedFileActionOK, "%s %s", cmd, path)
+	if err!= nil{
+		return nil, err
+	}
+	lines := strings.Split(msg, "\n")
```
Comment (corny) [inline ftp.go]: What happens if the filename contains spaces?

File: ftp.go (hunk 1 above)
Comment (zhaoya881010) [reply ftp.go]: ok

File: ftp.go (hunk 2 above)
Comment (zhaoya881010) [reply ftp.go]: it's fine work contains spaces.

**Comment by zhaoya881010** [comment]
@corny Fixed.

File: ftp.go [hunk 3]
```diff
@@ -535,6 +535,28 @@ func (c *ServerConn) List(path string) (entries []*Entry, err error) {
...
+func (c *ServerConn) Stat(path string)(entry *Entry, err error){
+	var cmd string
+	var parser parseFunc
+
+	if c.mlstSupported {
+		cmd = "MLST"
+		parser = parseRFC3659ListLine
+	} else {
+		return nil, errors.New("MLST unsupported")
+	}
```
Comment (corny) [inline ftp.go]: Please change that block to:
```golang
	if !c.mlstSupported {
		return nil, errors.New("MLST unsupported")
	}
```
And avoid the `cmd` and `parser` variables.

Comment (zhaoya881010) [reply ftp.go]: ok

**Comment by adamdecaf** [comment]
If I rebased this PR and revived it? Having `Stat()` would be pretty helpful.

**Comment by zhaoya881010** [comment]
support this feature ?

**Pull request closed without merging**